runtime/rasdump/dmpsup.c

```diff
@@ -714,6 +740,11 @@ configureDumpAgents(J9JavaVM *vm)
 	}
 	j9mem_free_memory(agentOpts);
 
+	/* Free optionString memory at original address for a mappped tool dump */
+	if (NULL != optionStringOriginal) {
+		j9mem_free_memory(optionStringOriginal);
```

Comment:
Arrange to mark `agentOpts.flags` with `J9RAS_DUMP_OPT_ARGS_ALLOC` so any number of the new options will be properly freed above.